### DISCUSSION TITLE
Looking for stable version of CTRLR for Win/OSX…

### DISCUSSION THREAD

**Discussion by 14bitMIDI** [RomanKubiak/ctrlr#472] [General]
Hello gang!
For quite some time I was happy with the old stable version of CTRLR for my Windows and OSX builds but as you all know time goes by and with the new OSX gatekeeper I can’t even develop for OSX anymore with the old CTRLR. It is not even opening on the newer M1 Macs. For the Windows I’m stuck with the VST2.4 but since Steinberg is ditching the VST2 format in favour of VST3 I am looking for a fresh alternative here. 
From some other post here I am understanding that there is a way to get CTRLR working on the current OS systems but it need to be compiled using Xcode or Visual Studio so there are no errors when exporting to a VST3 or AU format. (please correct me if I am wrong).
Is here anyone willing to help me out with this? I will be able to pay for your time of course. All I need is a standalone version of CTRLR that I can use to develop on the current Windows and OSX. Even one of the OSes would be fine for now if both are not possible or you are familiar more with one than the other but even that would be of great help to me! 
I hope someone will be able to chime in and we could discuss the details (my mail: karolobara(at)icloud.com).
THANK YOU IN ADVANCE!

**Comment by dobo365** [comment]
FYI, I'm still developing using 5.3.201 then producing OSX versions with 5.3.198 without any issue for M1 BS or Monterey users.
Otherwise, under code you can find the new latest one. I don't know if it runs fine but I think some of us are using it.

**Reply by 14bitMIDI** [reply]
Hello dobo365, thank you for your reply and sorry for my late reply - family vacation ;)
I want to ask you what OS version are you on? I have troubles even opening Ctrlr on my new Mac because of the gatekeeper not letting me to do it let alone developing anything on it. Any idea what I am doing wrong? Thanks!

**Reply by 14bitMIDI** [reply]
.... hello Mr.Goodweather ;) - learned so much from you on the old forum! Thank You!

**Comment by Godlike-Productions** [comment]
If you want VST3, the only option is 5.6.23, and you will need to recompile it to bake in the plugin name and manufacturer. The binaries are available here - https://godlike.com.au/index.php?id=361

If you are building VST3 with this, I recommend developing with 6.0.26 which is more stable.  5.6.23 will crash without showing why, unless you manually invoke functions in protected space using Assert. Once code is working, test it in 5.6.23 and then render out your VST3's

If you just want AU, standalone and VST2, then 5.3.201/5.3.198 is the way to go.

The main differences are that 5.6.23 and 6.0.26 require an extra option "\n" at the end of writeFileAsText and they won't allow tables to be used as inputs to make MIDI messages.  You need to cast them as memoryBlocks first (or convert to strings)

**Reply by 14bitMIDI** [reply]
Hi Godlike! A fan of yours here. Even bought few of your editors back in the day! Sorry for late reply but my family forced me to do some free time on the beach. 
Can you please elaborate on "...unless you manually invoke functions in protected space using Assert..." - that part is not clear to me at all. I will give it a shot but as already stated I am fearing any recompiling (maybe it is not that hard, who knows) so I will go back to the original post regarding recompiling Ctrlr using Visual Studio and I am sure more questions will pop up really fast - I hope you will be somehow able to guide me a bit through the process. Thank you for the feedback and I hope we can keep in touch!

**Reply by Godlike-Productions** [reply]
Here's an entry I put in the wiki on this.

https://github.com/damiensellier/CtrlrX/wiki/Calling-Protected-Functions
